### DISCUSSION TITLE
Graph display issue with/without filter

### DISCUSSION THREAD

**Discussion by dietybright** [akvorado/akvorado#1464] [Q&A]
When I choose time range in 10mins, without filtering, graph is not show specifc AS-X which consume high bw/volume, but with filtering for specific provider, it shows AS-X higher than the other.

When I run query on clickhouse-client with the filtering, it returns top 5 flows each 1s interval which contain AS-X with high traffic volume.

WITH
 source AS (SELECT * FROM flows SETTINGS asterisk_include_alias_columns = 1),
 rows AS (SELECT SrcAS, DstAS, ExporterName FROM source WHERE TimeReceived BETWEEN toDateTime('2024-10-31 04:00:00', 'UTC') AND toDateTime('2024-10-31 04:02:00', 'UTC') AND (InIfProvider = 'true' OR OutIfProvider = 'true') GROUP BY SrcAS, DstAS, ExporterName ORDER BY SUM(Bytes) DESC LIMIT 5)
SELECT 1 AS axis, * FROM (
SELECT
 toStartOfInterval(TimeReceived + INTERVAL 1 second, INTERVAL 1 second) - INTERVAL 1 second AS time,
 SUM(Bytes*SamplingRate*8)/1 AS xps,
 if((SrcAS, DstAS, ExporterName) IN rows, [concat(toString(SrcAS), ': ', dictGetOrDefault('asns', 'name', SrcAS, '???')), concat(toString(DstAS), ': ', dictGetOrDefault('asns', 'name', DstAS, '???')), ExporterName], ['Other', 'Other', 'Other']) AS dimensions
FROM source
WHERE TimeReceived BETWEEN toDateTime('2024-10-31 04:00:00', 'UTC') AND toDateTime('2024-10-31 04:02:00', 'UTC') AND (InIfProvider = 'true' OR OutIfProvider = 'true')
GROUP BY time, dimensions
ORDER BY time WITH FILL
 FROM toDateTime('2024-10-31 04:00:00', 'UTC')
 TO toDateTime('2024-10-31 04:02:00', 'UTC') + INTERVAL 1 second
 STEP 1
 INTERPOLATE (dimensions AS ['Other', 'Other', 'Other']))
//Return
467. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚    9003008 â”‚ ['132061: Realmove Company','omitted','omitted.local']  â”‚
468. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚  408051712 â”‚ ['omitted','17552: True Internet','omitted.local']      â”‚
469. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚   13049856 â”‚ ['Other','Other','Other']                                             â”‚
470. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚  495476736 â”‚ ['omitted','132618: True Move Company','omitted.local'] â”‚
471. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚  920502272 â”‚ ['omitted','132061: Realmove Company','omitted.local']  â”‚
472. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚   34734080 â”‚ ['omitted','132618: True Move Company','omitted.local'] â”‚
473. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚     425984 â”‚ ['Other','Other','Other']                                             â”‚
474. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚   11894784 â”‚ ['omitted','17552: True Internet','omitted.local']      â”‚
475. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚   16883712 â”‚ ['Other','Other','Other']                                             â”‚
476. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚   13860864 â”‚ ['132061: Realmove Company','omitted','omitted.local']  â”‚
477. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚   34357248 â”‚ ['omitted','7470: TRUE INTERNET','omitted.local']       â”‚
478. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚  495321088 â”‚ ['omitted','132618: True Move Company','omitted.local'] â”‚
479. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚ 1172086784 â”‚ ['omitted','132061: Realmove Company','omitted.local']  â”‚
480. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚  521453568 â”‚ ['omitted','17552: True Internet','omitted.local']      â”‚

But when I run query on clickhouse-client without the filtering, it returns top 5 flows each 1s interval which does not contain AS-X with higher volume than the others
WITH
 source AS (SELECT * FROM flows SETTINGS asterisk_include_alias_columns = 1),
 rows AS (SELECT SrcAS, DstAS, ExporterName FROM source WHERE TimeReceived BETWEEN toDateTime('2024-10-31 04:00:00', 'UTC') AND toDateTime('2024-10-31 04:02:00', 'UTC') GROUP BY SrcAS, DstAS, ExporterName ORDER BY SUM(Bytes) DESC LIMIT 5)
SELECT 1 AS axis, * FROM (
SELECT
 toStartOfInterval(TimeReceived + INTERVAL 1 second, INTERVAL 1 second) - INTERVAL 1 second AS time,
 SUM(Bytes*SamplingRate*8)/1 AS xps,
 if((SrcAS, DstAS, ExporterName) IN rows, [concat(toString(SrcAS), ': ', dictGetOrDefault('asns', 'name', SrcAS, '???')), concat(toString(DstAS), ': ', dictGetOrDefault('asns', 'name', DstAS, '???')), ExporterName], ['Other', 'Other', 'Other']) AS dimensions
FROM source
WHERE TimeReceived BETWEEN toDateTime('2024-10-31 04:00:00', 'UTC') AND toDateTime('2024-10-31 04:02:00', 'UTC')
GROUP BY time, dimensions
ORDER BY time WITH FILL
 FROM toDateTime('2024-10-31 04:00:00', 'UTC')
 TO toDateTime('2024-10-31 04:02:00', 'UTC') + INTERVAL 1 second
 STEP 1
 INTERPOLATE (dimensions AS ['Other', 'Other', 'Other']))
//Return
707. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚   502349208 â”‚ ['4058: CITIC Telecom International CPC','133210: EN Technologies','omitted'] â”‚
708. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚    26476968 â”‚ ['omitted','1221: Telstra Corporation','omitted']                   â”‚
709. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚  5976531504 â”‚ ['Other','Other','Other']                                                                  â”‚
710. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚     9423952 â”‚ ['omitted','13335: Cloudflare','omitted']                           â”‚
711. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚       42048 â”‚ ['omitted','4788: Telekom Malaysia Berhad (TM)','omitted']          â”‚
712. â”‚    1 â”‚ 2024-10-31 04:01:58 â”‚    15486704 â”‚ ['omitted','13335: Cloudflare','omitted']                           â”‚
713. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚   197969352 â”‚ ['4058: CITIC Telecom International CPC','133210: EN Technologies','omitted'] â”‚
714. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚    30778192 â”‚ ['omitted','1221: Telstra Corporation','omitted']                   â”‚
715. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚    13069320 â”‚ ['omitted','13335: Cloudflare','omitted']                           â”‚
716. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚    18415824 â”‚ ['omitted','13335: Cloudflare','omitted']                           â”‚
717. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚      104864 â”‚ ['omitted','4788: Telekom Malaysia Berhad (TM)','omitted']          â”‚
718. â”‚    1 â”‚ 2024-10-31 04:01:59 â”‚  5417542240 â”‚ ['Other','Other','Other']                                                                  â”‚
719. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚  8344718456 â”‚ ['Other','Other','Other']                                                                  â”‚
720. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚     7083616 â”‚ ['omitted','13335: Cloudflare','omitted']                           â”‚
721. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚   147574072 â”‚ ['4058: CITIC Telecom International CPC','133210: EN Technologies','omitted'] â”‚
722. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚       46752 â”‚ ['omitted','4788: Telekom Malaysia Berhad (TM)','omitted']          â”‚
723. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚     8798728 â”‚ ['omitted','13335: Cloudflare','omitted']                           â”‚
724. â”‚    1 â”‚ 2024-10-31 04:02:00 â”‚    61452640 â”‚ ['omitted','1221: Telstra Corporation','omitted']                   â”‚

Our routers have mix of sampling rate 1 and 1024. Will it causing issue with "SUM(Bytes*SamplingRate*8)/1 AS xps"?
Any thought?

**Comment by vincentbernat** [comment]
Check the output of the subrequest "rows" (and enclose code in triple backticks to make it easier to read, you can insert them with the `<>` icon). But yes, the `SUM(Bytes*SamplingRate*8)/1` is missing from the `rows` subrequest. Check if you it works when you change that.

**Reply by dietybright** [reply]
It seems corrected now, anyway to correct on GUI graph?
```
WITH source AS
    (
        SELECT *
        FROM flows
        SETTINGS asterisk_include_alias_columns = 1
    )
SELECT
    SrcAS,
    DstAS,
    ExporterName,
    SUM((Bytes * SamplingRate) * 8) / 1 AS total_traffic_bps
FROM source
WHERE (TimeReceived >= toDateTime('2024-10-31 04:00:00', 'UTC')) AND (TimeReceived <= toDateTime('2024-10-31 04:10:00', 'UTC'))
GROUP BY
    SrcAS,
    DstAS,
    ExporterName
ORDER BY total_traffic_bps DESC
LIMIT 5

Query id: 0f3ea114-639b-413d-8f32-69bd9763de75

   â”Œâ”€â”€SrcASâ”€â”¬â”€â”€DstASâ”€â”¬â”€ExporterNameâ”€â”€â”€â”€â”€â”€â”¬â”€total_traffic_bpsâ”€â”
1. â”‚ omitted â”‚ 131445 â”‚ SG2-omitted.local â”‚      473098756096 â”‚
2. â”‚ omitted â”‚ 131445 â”‚ SG1-omitted.local â”‚      291892731904 â”‚
3. â”‚ omitted â”‚ 132061 â”‚ SG1-omitted.local â”‚      231095549952 â”‚
4. â”‚ 133210 â”‚   9506 â”‚ SG1-omitted.local â”‚      211172024320 â”‚
5. â”‚ omitted â”‚ 131445 â”‚ SG1-omitted.local â”‚      178117369856 â”‚
```
```
WITH source AS
    (
        SELECT *
        FROM flows
        SETTINGS asterisk_include_alias_columns = 1
    )
SELECT
    SrcAS,
    DstAS,
    ExporterName,
    SUM((Bytes * SamplingRate) * 8) / 1 AS total_traffic_bps
FROM source
WHERE ((TimeReceived >= toDateTime('2024-10-31 04:00:00', 'UTC')) AND (TimeReceived <= toDateTime('2024-10-31 04:10:00', 'UTC'))) AND ((InIfProvider = 'true') OR (OutIfProvider = 'true'))
GROUP BY
    SrcAS,
    DstAS,
    ExporterName
ORDER BY total_traffic_bps DESC
LIMIT 5

Query id: d0473830-b968-4a70-9f28-600e799449a4

   â”Œâ”€â”€SrcASâ”€â”¬â”€â”€DstASâ”€â”¬â”€ExporterNameâ”€â”€â”€â”€â”€â”€â”¬â”€total_traffic_bpsâ”€â”
1. â”‚ omitted â”‚ 132061 â”‚ SG1-omitted.local â”‚      231095549952 â”‚
2. â”‚ omitted â”‚ 132618 â”‚ SG1-omitted.local â”‚      167920287744 â”‚
3. â”‚ omitted â”‚  17552 â”‚ SG1-omitted.local â”‚      149506695168 â”‚
4. â”‚ 132061 â”‚ omitted â”‚ SG1-omitted.local â”‚        4387774464 â”‚
5. â”‚  17552 â”‚ omitted â”‚ SG1-omitted.local â”‚        4071669760 â”‚
   â””â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜
```

**Reply by vincentbernat** [reply]
I have pushed a fix. In `docker/versions.yaml`, you can replace `akvorado:1.11.1` with `akvorado:main`, you should be able to test it once this run is complete: https://github.com/akvorado/akvorado/actions/runs/11607432606

**Reply by dietybright** [reply]
Hi Vincent,

It is fixed now, appreciated.

**Reply by kronicklez** [reply]
Hi @vincentbernat 

May i know whether the latest version 1.11.2 has solved this "Graph display issue with/without filter" issue? I'm still use version 1.11.1.

Thanks and appreciate your feedback

**Reply by vincentbernat** [reply]
Yes it is fixed.
